prody/proteins/interactions.py

```diff
@@ -0,0 +1,2034 @@
+    >> interactions = listLigandInteractions(calculations) """
+
+    try:
+        coords = (atoms._getCoords() if hasattr(atoms, '_getCoords') else
+                    atoms.getCoords())
+    except AttributeError:
+        try:
+            checkCoords(coords)
+        except TypeError:
+            raise TypeError('coords must be an object '
+                            'with `getCoords` method')
+
+    if not isinstance(interactions, list):
+        raise TypeError('interactions must be a list of interactions.')
+    
+    try:
+        output = kwargs['output']
+    except:
+        output = atoms.getTitle()+'_interaction.tcl'
+    
+    tcl_file = open(output, 'w') 
+    
+    if len(interactions[0]) >= 10: 
+        dic_color = {'hbond':'blue','pistack':'green','saltbridge':'yellow','pication':'orange',
+                     'hydroph_interaction':'silver','waterbridge':'cyan'}
+        
+        for i in interactions:
+            tcl_file.write('draw color '+dic_color[i[0]]+'\n')
+            
+            if i[0] == 'waterbridge':
+                hoh_id = atoms.select('x `'+str(i[11][0])+'` and y `'+str(i[11][1])+'` and z `'+str(i[11][2])+'`').getResnums()[0]
+                tcl_file.write('draw line {'+str(' '.join(map(str,i[9])))+'} {'+str(' '.join(map(str,i[11])))+'} style dashed width 4\n')
+                tcl_file.write('draw line {'+str(' '.join(map(str,i[10])))+'} {'+str(' '.join(map(str,i[11])))+'} style dashed width 4\n')
+                tcl_file.write('mol color Name\n')
+                tcl_file.write('mol representation Licorice 0.100000 12.000000 12.000000\n')
+                tcl_file.write('mol selection (resname '+i[1][:3]+' and resid '+str(i[1][3:])
+                               +' and chain '+i[3]+' and noh) or (resname '+i[4][:3]+' and resid '
+                               +str(i[4][3:])+' and chain '+i[6]+' and noh) or (water and resid '+str(hoh_id)+')\n')
+                
+            else:
+                tcl_file.write('draw line {'+str(' '.join(map(str,i[9])))+'} {'+str(' '.join(map(str,i[10])))+'} style dashed width 4\n')
+                tcl_file.write('mol color Name\n')
+                tcl_file.write('mol representation Licorice 0.100000 12.000000 12.000000\n')
+                tcl_file.write('mol selection (resname '+i[1][:3]+' and resid '+str(i[1][3:])
+                               +' and chain '+i[3]+' and noh) or (resname '+i[4][:3]+' and resid '
+                               +str(i[4][3:])+' and chain '+i[6]+' and noh)\n')
+            tcl_file.write('mol material Opaque\n')
+            tcl_file.write('mol addrep 0 \n')            
+
+
+    tcl_file.write('draw materials off')
+    tcl_file.close()   
+    LOGGER.info("TCL file saved")
+
+
+class Interactions(object):
+
+    """Class for Interaction analysis of proteins."""
+
+    def __init__(self, title='Unknown'):
+        self._title = str(title).strip()
+           
+    def setTitle(self, title):
```

Comment:
why does setTitle initialise all these things? It's not even used by any other function. It would be much better to do this in __init__ or calcProteinInteractions. Or perhaps, you could call setTitle from calcProteinInteractions

Reply:
Moved to init